AutomatedLab/AutomatedLabSQL.psm1

```diff
@@ -248,50 +248,15 @@ GO
 
                 New-LabSqlAccount -Machine $machine -RoleProperties $role.Properties
 
-                $scriptBlock = {
-                    Write-Verbose 'Installing SQL Server...'
-
-                    $dvdDrive = ''
-                    $startTime = (Get-Date)
-                    while (-not $dvdDrive -and (($startTime).AddSeconds(120) -gt (Get-Date)))
-                    {
-                        Start-Sleep -Seconds 2
-                        $dvdDrive = (Get-WmiObject -Class Win32_CDRomDrive | Where-Object MediaLoaded).Drive
-                    }
-
-                    if ($dvdDrive)
-                    {
-                        #Configure App Compatibility for SQL Server 2008. Otherwise a warning pop-up will stop the installation
-                        New-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags' -Name '{f2d3ae3a-bfcc-45e2-bf63-178d1db34294}' -Value 4 -PropertyType 'DWORD' -ErrorAction SilentlyContinue
-                        New-ItemProperty -Path 'HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\AppCompatFlags' -Name '{45da5a8b-67b5-4896-86b7-a2e838aee035}' -Value 4 -PropertyType 'DWORD' -ErrorAction SilentlyContinue
-
-                        $installation = Start-Process -FilePath "$dvdDrive\Setup.exe" -ArgumentList $setupArguments -Wait -LoadUserProfile -PassThru
-
-                        if ($installation.ExitCode -notin 0,3010)
-                        {
-                            throw "SQL Setup failed with exit code $($installation.ExitCode)"
-                        }
-
-                        Write-Verbose 'SQL Installation finished. Restarting machine.'
-
-                        Restart-Computer -Force
-                    }
-                    else
-                    {
-                        Write-Error -Message 'Setup.exe in ISO file could not be found (or ISO was not successfully mounted)'
-                    }
-                }
-
                 $param = @{}
                 $param.Add('ComputerName', $machine)
-                $param.Add('ActivityName', 'Install SQL Server')
+                $param.Add('LocalPath', "$($dvdDrive.DriveLetter)\Setup.exe")
                 $param.Add('AsJob', $true)
                 $param.Add('PassThru', $true)
                 $param.Add('NoDisplay', $true)
-                $param.Add('Scriptblock', $scriptBlock)
-                $param.Add('Variable', (Get-Variable -Name setupArguments))
-
-                $jobs += Invoke-LabCommand @param
+                $param.Add('CommandLine', $setupArguments)
+                $param.Add('ExpectedReturnCodes', (0,3010))
+                $jobs += Install-LabSoftwarePackage @param -UseShellExecute
```

Comment:
Maybe I'm missing it @raandree , but where is the configuration file copied to the remote system if a config file is being used?

Reply:
This is handled in line 217 to 240. I have not changed that.

Reply:
I don't think so. Since you tested it, I don't see a reason why it would not work.